src/providers/irc/AbstractIrcServer.cpp

```diff
@@ -23,6 +23,19 @@ AbstractIrcServer::AbstractIrcServer()
     this->writeConnection_->moveToThread(
         QCoreApplication::instance()->thread());
 
+    // Apply a leaky bucket rate limiting to JOIN messages
+    auto actuallyJoin = [&](QString message) {
+        if (!this->channels.contains(message))
+        {
+            qCDebug(chatterinoIrc)
+                << QString("attempted to join %1 but it does not exist anymore")
+                       .arg(message);
+            return;
+        }
+        this->readConnection_->sendRaw("JOIN #" + message);
+    };
+    this->joinBucket_.reset(new RatelimitBucket(18, 10500, actuallyJoin, this));
```

Comment:
Can we turn these magic numbers into named constants?

Reply:
Yes, I do think so. See 658f28f31eb771ca7fcd156b524de5addb8c3072